### PR TITLE
libafl_qemu_sys and libafl_qemu_build to have bindgen with QEMU

### PR DESCRIPTION
**Pull request by andreafioraldi**
Mainly to access the inner structures of QEMU in Rust in libafl_qemu

### CODE REVIEW

**Comment by domenukk** [comment]
somehow this breaks with
```bash

  --- stderr
  make: *** [Makefile:165: run-ninja] Error 1
```

File: libafl_qemu/src/lib.rs [hunk 1]
```diff
@@ -14,9 +14,14 @@
 #![allow(clippy::borrow_deref_ref)]
 // Allow only ATM, it will be evetually removed
 #![allow(clippy::missing_safety_doc)]
+// libafl_qemu_sys export types with mepty struct markers (e.g. struct {} start_init_save)
```
Comment (domenukk) [inline libafl_qemu/src/lib.rs:17]: *empty

File: libafl_qemu/src/lib.rs [hunk 2]
```diff
@@ -14,9 +14,14 @@
...
+// This causes bindgen to generate empty Rust struct that are generally not FFI-safe due to C++ having empty structs with size 1
+// As the QEMU codebase is C, it is FFI-safe and we just ignore the warning
+#![allow(improper_ctypes)]
```
Comment (domenukk) [inline libafl_qemu/src/lib.rs:20]: Why not set the allow only for libafl_qemu_sys, instead of the whole crate?

File: libafl_qemu/src/emu.rs [hunk 3]
```diff
@@ -16,23 +16,73 @@ use num_enum::{IntoPrimitive, TryFromPrimitive};
...
+
+    #[must_use]
+    pub fn decode_from(encoded: u64) -> (Self, u32) {
+        let low = (encoded & 0xFFFFFFFF) as u32;
+        let high = (encoded >> 32) as u32;
+        (Self { oi: high }, low)
+    }
+
+    #[must_use]
+    pub fn new(oi: libafl_qemu_sys::MemOpIdx) -> Self {
```
Comment (domenukk) [inline libafl_qemu/src/emu.rs:72]: you could consider
```rust
impl From<MemOpIdx> for MemAccessInfo {...
```

**Review by domenukk — APPROVED** [review]
Looks super sweet!

File: libafl_qemu/src/lib.rs (hunk 2 above)
Comment (andreafioraldi) [reply libafl_qemu/src/lib.rs:20]: Because I'm using types of the sys in FFI code in libafl_qemu.

**Pull request merged**